pkgs/development/libraries/caliper/default.nix

```
@@ -0,0 +1,96 @@
+{ lib, stdenv, fetchurl, python3, cmake, adiak, papi
+, withCuda ? false, cudatoolkit
+, withRocm ? false, rocm-runtime
+, withFortran? false, gfortran
+, withOpenmpi? false, openmpi
+, withMpich ? false, mpich
+, libunwindSupport ? true, libunwind
+, libpfmSupport ? true, libpfm
+, elfutilsSupport ? true, elfutils
+, shared ? !stdenv.hostPlatform.isStatic
+}:
+
+# https://github.com/spack/spack/blob/develop/var/spack/repos/builtin/packages/caliper/package.py
+
+let
+   onOffBool = b: if b then "ON" else "OFF";
+   withMPI = (withMpich == true || withOpenmpi == true);
+in
+
+# We can't have both mpis being used (is this the right way to do this?)
+assert (withMpich && !withOpenmpi) || (!withMpich && withOpenmpi) || (!withMpich && !withOpenmpi);
+
+stdenv.mkDerivation rec {
+  pname = "caliper";
+  version = "2.8.0";
+
+  src = fetchurl {
+    url = "https://github.com/LLNL/Caliper/archive/v${version}.tar.gz";
+    sha256 = "17807b364b5ac4b05997ead41bd173e773f9a26ff573ff2fe61e0e70eab496e4";
+  };
+
+  meta = with lib; {
+    description = "Program instrumentation and performance measurement framework";
+    longDescription = ''
+      A program instrumentation and performance measurement
+      framework. It is designed as a performance analysis toolbox in a
+      library, allowing one to bake performance analysis capabilities
+      directly into applications and activate them at runtime.
+    '';
+    homepage = "https://github.com/LLNL/Caliper";
+    license = licenses.bsd3;
+    maintainers = [ maintainers.vsoch ];
+    platforms = platforms.linux;
+  };
+
+  # This means provided natively in Linux I think?
+  nativeBuildInputs = [ ];
+
+  # We can eventually add these variants if needed
+  # depends_on("sosflow@spack", when="@1.0:1+sosflow")
+  # variant("gotcha", default=is_linux, description="Enable GOTCHA support")
+  # variant("sampler", default=is_linux, description="Enable sampling support on Linux")
+  # variant("sosflow", default=False, description="Enable SOSflow support")
+  # variant("fortran", default=False, description="Enable Fortran support")
+
+  buildInputs = [ cmake python3 papi adiak ] ++
+    lib.optional libpfmSupport libpfm ++
+    lib.optional libunwindSupport libunwind ++
+    lib.optional elfutilsSupport elfutils ++
+
+    # These shouldn't be both provided
+    lib.optional withOpenmpi openmpi ++
+    lib.optional withMpich mpich;
```

```suggestion
  buildInputs = [ cmake python3 papi adiak ]
    ++ lib.optional libpfmSupport libpfm
    ++ lib.optional libunwindSupport libunwind
    ++ lib.optional elfutilsSupport elfutils
    ++ lib.optional withOpenmpi openmpi
    ++ lib.optional withMpich mpich;
```
this should be asserted